### PR TITLE
make.js script and Node 16 capability #3438

### PR DESCRIPTION
**Pull request by DmitriiBobreshev**
**Task name**: none

**Description**: 
To be able to update make.js script to use node 16 we need to upgrade common packages on ts 4.0.2 otherwise it will be
types conflict because when we are using ts version 2 we using our own types, and for ts version 4 we installing packages, so when we compiling
shared packages there is a problem that ts finding not only types from shared packages(common-npm-packages) but also from root node_modules folder,
so that's why the following changes were made:

- Deleted docker-common common package because it's not using in any other tasks
- Migrated from azure-pipelines-tasks-azure-arm-rest to azure-pipelines-tasks-azure-arm-rest-v2 in JenkinsDownloadArtifacts
- Deleted azure-pipelines-tasks-azure-arm-rest because it's not using in any other tasks
- Added capability to be able to run tests and builds tasks under node16
- Upgraded ts and types to use it with node 6, 10 and 16 in the roots package.json
- remove typed.s for mocha, semver, q, not etc. in some tests because we don't need it after upgrade make.js to ts 4
- Removed utility-common and migrated all tasks which using it on utility-common-v2 because they are basically the same

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Checklist**:
- [x] Task version was bumped - Yes for JenkinsDownloadArtifacts
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: make.js [hunk 1]
```diff
@@ -341,6 +341,7 @@ CLI.test = function(/** @type {{ suite: string; node: string; task: string }} */
             return;
         }
 
+        console.log(nodeVersion)
```
Comment (KonstantinTyukalov) [inline make.js]:
```suggestion
        console.log(`Node version: ${nodeVersion}`)
```

**Review by KonstantinTyukalov — APPROVED** [review]

File: make.js (hunk 1 above)
Comment (DmitriiBobreshev) [reply make.js]: sorry, accidentally added this line for debugging, already removed it

**Review by DenisRumyantsev — APPROVED** [review]

File: make-util.js [hunk 2]
```diff
@@ -331,6 +331,9 @@ exports.ensureTool = ensureTool;
 
 var installNode = function (nodeVersion) {
     switch (nodeVersion || '') {
+        case '16':
+            nodeVersion = 'v16.15.1';
+            break;
         case '14':
             nodeVersion = 'v14.10.1';
             break;
```
Comment (DenisRumyantsev) [inline make-util.js:339]: Please remove it
suggestion```
```

**Review by kirill-ivlev — APPROVED** [review]

**Review by mmrazik — APPROVED** [review]

**Comment by DmitriiBobreshev** [comment]
Closing the PR.
The PR will be separated into 2 different PRs to reduce file count.
1) to remove common packages wich are not using
2) to migrate make.js to Node16

**Pull request closed without merging**